service/src/main/java/org/jboss/sbomer/k8s/TaskRunStatusHandler.java

```diff
@@ -0,0 +1,214 @@
+                    "Found a Tekton TaskRun without the SBOM id label: '{}', skipping, but this is not good!",
+                    taskRun.getMetadata().getName());
+            return false;
+        }
+
+        // TaskRun does not have proper status yet, nothing to update, skipping
+        if (taskRun.getStatus() == null || taskRun.getStatus().getConditions() == null
+                || taskRun.getStatus().getConditions().isEmpty()) {
+            log.debug("Found Tekton TaskRun without status ready: '{}', skipping", taskRun.getMetadata().getName());
+            return false;
+        }
+
+        log.debug("TaskRun '{}' is valid for processing", taskRun.getMetadata().getName());
+
+        return true;
+    }
+
+    @Transactional
+    protected void updateStatus(String sbomId, SbomStatus status) {
+
+        SbomStatus cachedStatus = statusCache.get(sbomId);
+
+        // If our cache has different content this means that we need to update the resource.
+        // First update is always performed and the cache is populated with it.
+        if (cachedStatus == status) {
+            log.debug("Skipping update for Sbom id '{}' because new status is the same as old: '{}'", sbomId, status);
+            return;
+        }
+
+        Sbom sbom = sbomRepository.findById(Long.valueOf(sbomId));
+
+        if (sbom == null) {
+            log.warn("Could not find Sbom id '{}', skipping updating the status", sbomId);
+            return;
+        }
+
+        // Update resource
+        sbom.setStatus(status);
+        // Update status
+        statusCache.put(sbomId, status);
+        // Save the resource
+        sbomRepository.saveSbom(sbom);
+
+        log.info("Updated Sbom id '{}' with status: '{}'", sbomId, status);
+
+    }
+
+    protected SbomStatus toStatus(TaskRun taskRun) {
+        // Find last update
+        String taskRunStatus = Optional.ofNullable(taskRun.getStatus().getConditions())
+                .map(Collection::stream)
+                .orElseGet(Stream::empty)
+                .findFirst()
+                .orElse(null)
+                .getStatus();
+
+        SbomStatus status = null;
+
+        // Set the status based on the
+        switch (taskRunStatus) {
+            case "Unknown":
+                status = SbomStatus.IN_PROGRESS;
+                break;
+            case "True":
+                status = SbomStatus.READY;
+                break;
+            case "False":
+                status = SbomStatus.FAILED;
+                break;
+            default:
+                log.error("Received unknown status from TaskRun: '{}'", taskRunStatus);
+                return null;
+        }
+
+        return status;
+    }
+
+    protected void handleTaskRunUpdate(TaskRun taskRun) {
+        if (!isUpdateable(taskRun)) {
+            return;
+        }
+
+        SbomStatus status = toStatus(taskRun);
+
+        updateStatus(taskRun.getMetadata().getLabels().get(Constants.TEKTON_LABEL_SBOM_ID), status);
```

Comment:
Do we want to execute this if status is null? (could be returned by `toStatus`)

Reply:
Done!